observability.go

```diff
@@ -124,16 +124,18 @@ func newLoggerProvider(ctx context.Context, config OtelConfig, serviceResource *
 	return loggerProvider, nil
 }
 
-func newServiceResource(ctx context.Context, name string) (*resource.Resource, error) {
+func newServiceResource(ctx context.Context, hostData HostData) (*resource.Resource, error) {
 	providerBinary, err := os.Executable()
 	if err != nil {
 		return nil, err
 	}
 
 	return resource.New(ctx,
 		resource.WithAttributes(
-			semconv.ServiceNameKey.String(filepath.Base(providerBinary)),
-			semconv.ServiceInstanceIDKey.String(name),
+			semconv.ServiceNameKey.String(hostData.ProviderKey),
+			semconv.HostIDKey.String(hostData.HostID),
+			semconv.ContainerIDKey.String(hostData.InstanceID),
```

Comment:
Let's use `ServiceInstanceIDKey` instead of `ContainerIDKey`

Reply:
Done